packages/material-ui/src/Radio/index.js

```diff
@@ -1 +1 @@
-export { default } from './Radio';
+export { default, RadioGroupContext } from './Radio';
```

Comment:
Would RadioGroupContext not be with RadioGroup?

Reply:
When using RadioGroup, you use Radio. When using Radio, you might not use RadioGroup. I had to choose a module to host the variable. I went with the location that maximize tree shaking.

Reply:
None of this should matter. Both components use the context so they import it anyway. `Radio` should import `RadioGroupContext` but not `RadioGroupContext` even if that is placed next to it. I agree with @joshwooding that the context should live next to the provider.

Reply:
If you are only importing Radio (a react final form or formik user might not need the RadioGroup) and doesn't have esm support, you will bundle RadioGroup for nothing. Am I missging something?

Reply:
Who doesn't have esm support? Can we rewrite all the demos to do import { A, B } from '@material-ui/core'?

Reply:
Who is not using RadioGroup with Radio? Maybe it doesn't make sense. Was I overthinking it?
Should we move the context to a different file like we do with the ThemeContext?

Reply:
Ok, let's be consistent. I'm gonna reproduce List, Theme and Table strategy.

Reply:
> If you are only importing Radio (a react final form or formik user might not need the RadioGroup) and doesn't have esm support, you will bundle RadioGroup for nothing. Am I missing something?

No. Unless `Radio.js` imports `import * as RadioGroup from '../RadioGroup'`. You probably would do `import { RadioGroupContext } from '../RadioGroup'` which shakes perfectly fine.

Reply:
What about the: ` doesn't have esm support`? If the user bundler doesn't support tree-shaking?

Reply:
Then why should we care if the user doesn't? We're talking about webpack 1 support here.

Reply:
So back to my question:
> Who doesn't have esm support? Can we rewrite all the demos to do import { A, B } from '@material-ui/core'?

Reply:
We could but we support the previous pattern as well. It was the whole point of #13391.

It becomes also very unwieldy with TypeScript. We would need to rewrite a lot of index files for no gain.

Reply:
> for no gain

Are you saying 1. and 2. are as good?

1.
```jsx
import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
```

2.
```jsx
import React from 'react';
import PropTypes from 'prop-types';
import {
  withStyles,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Paper,
} from '@material-ui/core';
```

#13391 is needed no matter what we document in the demos.

Reply:
> #13391 is needed no matter what we document in the demos.

Partially yes. We wouldn't need the `package.json` files if we would only support the second pattern

> Are you saying 1. and 2. are as good?

Apart from the syntax I don't know of any observable difference.

Reply:
The problem with the approach n°1 is that people duplicate the component name.
```jsx
import TableHead from '@material-ui/core/TableHead';
```
Have you never been hit by this mistake?
```jsx
import TableRow from '@material-ui/core/TableHead';
```

Reply:
I wasn't aware of the TypeScript issue.

Reply:
> import TableRow from '@material-ui/core/TableHead';

This is what types are for. All the rest is not a mistake because `module name !== implemented name`. 90% of the times I use interfaces instead of implementations. When I do care about implementation the imported name never matters.

Also: We were discussing were the context should live not named vs. default. We can discuss this if there's an issue. I will not engage in arguments with constructed issues.

Reply:
> Also: We were discussing were the context should live not named vs. default. We can discuss this if there's an issue. I will not engage in arguments with constructed issues.

I'm sorry, the discussion went off the rails. Regarding the context location. I have updated the pull request to follow the existing convention. We have a new file: `src/RadioGroup/RadioGroupContext.js`.